app/frontend/src/features/communities/PagePage.tsx

```diff
@@ -0,0 +1,62 @@
+import PageTitle from "../../components/PageTitle";
+import React, { useEffect, useState } from "react";
+import { useParams } from "react-router-dom";
+import { useAppDispatch } from "../../store";
+import { Page } from "../../pb/pages_pb"
+import { service } from "../../service";
+import Alert from "../../components/Alert";
+import CircularProgress from "../../components/CircularProgress";
+import TextBody from "../../components/TextBody";
+import { useHistory } from "react-router-dom";
+import { pageRoute } from "../../AppRoutes"
+import Markdown from "../../components/Markdown";
+
+export default function PagePage() {
+  const [loading, setLoading] = useState(false);
+  const [error, setError] = useState("");
+  const [page, setPage] = useState<Page.AsObject | null>(null);
+
+  const dispatch = useAppDispatch();
+  const history = useHistory();
+
+  const { pageId, pageSlug } = useParams<{ pageId: string, pageSlug?: string }>();
+
+  useEffect(() => {
+    if (!pageId) return;
+    (async () => {
+      setLoading(true);
+      try {
+        const page = await service.pages.getPage(Number(pageId))
+        setPage(page);
+        if (page.slug !== pageSlug) {
+          // if the address is wrong, redirect to the right place
+          history.push(`${pageRoute}/${page.pageId}/${page.slug}`);
+        }
+      } catch (e) {
+        console.error(e)
+        setError(e.message);
+      }
+      setLoading(false);
+    })();
+  }, [pageId]);
```

Comment:
To make the linter happy 

```suggestion
  }, [pageId, pageSlug, history]);
```